### DISCUSSION TITLE
Render Menu in a portal to show as an absolute realtive to menu button

### DISCUSSION THREAD

**Discussion by cubecleveland** [ariakit/ariakit#930] [Questions] [answered]
Hello,
I'm using the menu inside a container that has a scrolling and i want the menu to show absolute to the entirre viewport so activating the menu will not scroll the container when it goes out of bounds.

My though is to render the menu inside a portal and then pass the x,y of the button click to show  the menu.

Is there a best practice to acieve that? does anyone have an example using popper?

**Comment by diegohaz** [comment, marked as answer]
You can just wrap the `Menu` component with `Portal`:
```jsx
import { useMenuState, Menu, MenuItem, MenuButton } from "reakit/Menu";
import { Portal } from "reakit/Portal";

function Example() {
  const menu = useMenuState();
  return (
    <>
      <MenuButton {...menu}>Preferences</MenuButton>
      <Portal>
        <Menu {...menu} aria-label="Preferences">
          <MenuItem {...menu}>Settings</MenuItem>
          <MenuItem {...menu}>Extensions</MenuItem>
          <MenuItem {...menu}>Keyboard shortcuts</MenuItem>
        </Menu>
      </Portal>
    </>
  );
}

```
